projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -198,6 +198,10 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
         const selectedIndex = this.displayedListData.indexOf(eventArgs.checkbox.value);
         const selectAllBtn = this.displayedListData[0];
 
+        if (this.displayedListData[selectedIndex].isSelected === eventArgs.checked) {
```

Comment:
@damyanpetev Yes. When setting the initial state all checkboxes get checked, because the first checkbox enters as checked initially (the select all one).